### ISSUE TITLE
MacOS GitHub Action issue

### ISSUE THREAD

**Issue by Cyclic3** [Ciphey/Ciphey#238]
**Describe the bug**
We keep getting a segfault in the github action for ciphey, but this cannot be repro'd on anyone's real-world system

**To Reproduce**
1. Add MacOS to CI
2. ha ha CI machine goes brrr

**Expected behavior**
Tests to run fine, like they do on all other machines

**Comment by SR2374** [comment]
How i can fix

**Comment by Cyclic3** [comment]
Hi @SR2374 !

I'm not entirely sure how to fix the bug (or I would have already done it :smiley:). If you have some ideas you would like to try, try forking Ciphey, and fiddling with `.github/workflows/tests2.yml` there (macos is currently commented out in that file, and needs to be re-added to the list). 

If you do get it working, then please shoot us a pull request, and we will make sure to add you as a code contributor!

**Comment by SR2374** [comment]
Oki😊

**Comment by dreness** [comment]
Looking at the crash report, the crashed thread looks like:

```
Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   org.python.python             	0x000000010207f9a1 PyTuple_New + 241
1   org.python.python             	0x0000000102082479 PyType_Ready + 233
2   org.python.python             	0x0000000102082435 PyType_Ready + 165
3   _ciphey_core_py.so            	0x0000000101ec9664 PyInit__ciphey_core_py + 2212
4   org.python.python             	0x0000000101905ebe _PyImport_LoadDynamicModuleWithSpec + 613
```
Find `_ciphey_core_py.so` - something like this should do the trick:
```
find $(python -c 'import site; print(site.getsitepackages()[0])') -name "*ciphey*so"
```

Run `otool -L` passing the path to `_ciphey_core_py.so`. Mine is inside a virtualenv, but yours might not be.

```
❯ otool -L /Users/andre/.virtualenvs/py38/lib/python3.8/site-packages/cipheycore/_ciphey_core_py.so        
/Users/andre/.virtualenvs/py38/lib/python3.8/site-packages/cipheycore/_ciphey_core_py.so:
	@rpath/_ciphey_core_py.so (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/Python (compatibility version 3.8.0, current version 3.8.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 902.1.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)
```

ciphey is looking in the wrong place for the Python interpreter, because I have homebrew installed to `/opt/brew` instead of `/usr/local`. I can fix this with `install_name_tool`:

```
❯ install_name_tool -change /usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/Python /opt/brew/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/Python /Users/andre/.virtualenvs/py38/lib/python3.8/site-packages/cipheycore/_ciphey_core_py.so
```

All better:

```
❯ ciphey
Error. No inputs were given to Ciphey. 
Usage: ciphey [OPTIONS] [TEXT_STDIN]

  Ciphey - Automated Decryption Tool

  Documentation:  https://docs.ciphey.online

...
```

**Comment by Kapp1** [comment]
I lost my connection before finishing I think the last message was

pip install --upgrade pip

**Comment by bee-san** [comment]
This is being fixed in #591
